docs/src/index.md

```diff
@@ -1,12 +1,119 @@
-```@meta
-CurrentModule = MLJNearestNeighborsInterface
+# NearestNeighborModels - Docs
+
+NearestNeighborModels is a julia package providing implemtation of various 
+k-nearest-neighbor classifiers and regressors models for use with 
+[MLJ](https://alan-turing-institute.github.io/MLJ.jl/dev/) machine learning 
+framework. It also provides users with an array of weighting kernels to choose 
+from for prediction.
+
+NearestNeighborModels builds on Kristoffer Carlsson's 
+[NearestNeighbors](https://github.com/KristofferC/NearestNeighbors.jl) package(for 
+performing efficient nearest neighbor searches) and earlier contributions from Thibaut 
+Lienart originally residing in 
+[MLJModels.jl](https://github.com/alan-turing-institute/MLJModels.jl/blob/98618d7be53f72054de284fa1796c5292d9071bb/src/NearestNeighbors.jl#L1).
+
+
+# Installation
+
+On a Julia>=1.0 NearestNeighborModels can be added via Pkg
+as shown below.
+
+```julia
+using Pkg
+Pkg.add("NearestNeighborModels") 
+```
+
+# Usage
+
+To use any model implemented in this package, the model must first be wrapped in an MLJ 
+machine alongside the required data. Users also get additional features from MLJ including 
+performance evaluation, hyper-parameter tuning, stacking etc.
+The following example shows how to train a `KNNClassifier` on the crabs dataset.
+
+```julia
+using NearestNeighborModels, MLJ
+X, y = @load_crabs; # loads the crabs dataset from MLJ
+train_inds, test_inds = partition(1:nrows(X), 0.7, shuffle=false, rng=100);
```

Comment:
No point specifying `rng` if `shuffle=false` (the default). If you omit `shuffle` and specify an `rng` (integer or `AbstractRNG`) then shuffling is implied.